arrows/ocv/refine_detections_write_to_disk.cxx

```diff
@@ -171,31 +166,30 @@ ::refine( vital::image_container_sptr image_data,
 
     cv::Size s = img.size();
     vital::bounding_box_d bounds( vital::bounding_box_d::vector_type( 0, 0 ),
-      vital::bounding_box_d::vector_type( s.width, s.height ) );
+                                  vital::bounding_box_d::vector_type( s.width, s.height ) );
 
+    // Clip detection box to image bounds.
     bbox = intersection( bounds, bbox );
 
     // Generate output filename
-    std::string ofn;
-    size_t max_len = d_->pattern.size() + 4096;
-    ofn.resize( max_len );
-    int num_bytes = snprintf( &ofn[0], max_len, d_->pattern.c_str(), d_->id++,
-                                                bbox.upper_left()[0], bbox.upper_left()[1],
-                                                bbox.width(), bbox.height(), filename.c_str() );
-
-    if( num_bytes < 0 )
+    std::string ofn = kwiver::vital::string_format( d_->pattern,
+                      d_->id++, filename.c_str(),
```

Comment:
Maybe here, can filename remain unset?

Reply:
If filename is unset, this will be the null string. The filename may not be available from the attached metadata, so not clear what to use as a default. Existing code leaves it as an empty string.